pkg/controller/case_test.go

```diff
@@ -0,0 +1,583 @@
+func (ct *ControllerTest) CreateClusterServiceBrokerWithBasicAuth() {
+	csb := ct.createClusterServiceBroker()
+	csb.Spec.AuthInfo = &v1beta1.ClusterServiceBrokerAuthInfo{
+		Basic: &v1beta1.ClusterBasicAuthConfig{
+			SecretRef: &v1beta1.ObjectReference{
+				Name:      authSecretName,
+				Namespace: testNamespace,
+			},
+		},
+	}
+	ct.scInterface.ClusterServiceBrokers().Create(csb)
+}
+
+func (ct *ControllerTest) CreateServiceInstance() {
+	ct.scInterface.ServiceInstances(testNamespace).Create(&v1beta1.ServiceInstance{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: testServiceInstanceName,
+		},
+		Spec: v1beta1.ServiceInstanceSpec{
+			PlanReference: v1beta1.PlanReference{
+				ClusterServiceClassExternalName: testClassExternalID,
+				ClusterServicePlanExternalName:  testPlanExternalID,
+			},
+			ExternalID: testExternalID,
+			// Plan and Class refs are added by a Webhook, which is not tested here
+			ClusterServicePlanRef: &v1beta1.ClusterObjectReference{
+				Name: testPlanExternalID,
+			},
+			ClusterServiceClassRef: &v1beta1.ClusterObjectReference{
+				Name: testClassExternalID,
+			},
+		},
+	})
+}
+
+func (ct *ControllerTest) DeleteServiceInstance() {
+	ct.scInterface.ServiceInstances(testNamespace).Delete(testServiceInstanceName, &v1.DeleteOptions{})
+}
+
+func (ct *ControllerTest) CreateBinding() {
+	ct.scInterface.ServiceBindings(testNamespace).Create(&v1beta1.ServiceBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:  testNamespace,
+			Name:       testBindingName,
+			Generation: 1,                                         // set by the Webhook
+			Finalizers: []string{v1beta1.FinalizerServiceCatalog}, // set by the Webhook
+		},
+		Spec: v1beta1.ServiceBindingSpec{
+			InstanceRef: v1beta1.LocalObjectReference{
+				Name: testServiceInstanceName,
+			},
+			ExternalID: testServiceBindingGUID,
+			SecretName: testBindingName, // set by the webhook
+		},
+	})
+}
+
+func (ct *ControllerTest) DeleteBinding() {
+	ct.scInterface.ServiceBindings(testNamespace).Delete(testBindingName, &v1.DeleteOptions{})
+}
+
+func (ct *ControllerTest) CreateSecretWithBasicAuth(username, password string) {
+	ct.k8sClient.CoreV1().Secrets(testNamespace).Create(&corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: testNamespace,
+			Name:      authSecretName,
+		},
+		Data: map[string][]byte{
+			v1beta1.BasicAuthUsernameKey: []byte(username),
+			v1beta1.BasicAuthPasswordKey: []byte(password),
+		},
+	})
+}
+
+func (ct *ControllerTest) UpdateSecretWithBasicAuth(username, password string) {
+	ct.k8sClient.CoreV1().Secrets(testNamespace).Update(&corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: testNamespace,
+			Name:      authSecretName,
+		},
+		Data: map[string][]byte{
+			v1beta1.BasicAuthUsernameKey: []byte(username),
+			v1beta1.BasicAuthPasswordKey: []byte(password),
+		},
+	})
+}
+
+func (ct *ControllerTest) AssertClusterServiceClassAndPlan(t *testing.T) {
+	err := ct.WaitForClusterServiceClass()
+	if err != nil {
```

Comment:
Why u sometimes using the **testify** assert library and sometimes u are using the native approach?